packages/vite/src/node/server/middlewares/base.ts

```diff
@@ -7,14 +7,20 @@ import type { Connect } from 'types/connect'
 export function baseMiddleware({
   config
 }: ViteDevServer): Connect.NextHandleFunction {
-  const base = config.base
+  const { base, root } = config
 
   // Keep the named function. The name is visible in debug logs via `DEBUG=connect:dispatcher ...`
   return function viteBaseMiddleware(req, res, next) {
     const url = req.url!
     const parsed = parseUrl(url)
     const path = parsed.pathname || '/'
 
+    if (path.startsWith(root)) {
```

Comment:
root + base = `/home/xxx/home/` ?

In my test case, path will be `<root>/[relative path from root]`

Reply:
Oh, gotcha, you are actually trying to fix the other way around.
I thought the bug was that `/home/xxx/home/path` was converted into `/xxx/home/path`, that seems to be an issue currently. 
I don't understand your case, I think that absolute paths will be covered by the rule I proposed. And paths relative to the project root will also work.
The only case that won't be covered is is when you are trying to have an absolute path out of root, but in that case Vite will use `/@fs/`, no?

Reply:
I update the desc.

> If set the base URL to root on Linux and use vue plugin, vue plugin will import css by absolute path. And vite will replace base to `/`, and absolute path startWith `base path` will got a error.

Reply:
@patak-dev the case is

config with

```ts
{
  base: "/home/"
}
```

* had absolute path `/home/xxx`
* root is `/home/xxx/project`

so the request must be `/home/xxx/project/assets` (because the request will call by the vue plugin)

and then replace root to be `/assets`

Reply:
> I thought the bug was that /home/xxx/home/path was converted into /xxx/home/path, that seems to be an issue currently.

Is there a way to reproduce this possible? I've been thinking about it for a long time and can't figure it out.  😂